src/main/java/info/tehnut/soulshards/api/BindingEvent.java

```diff
@@ -1,15 +1,16 @@
 package info.tehnut.soulshards.api;
 
-import net.fabricmc.fabric.util.HandlerArray;
 import net.minecraft.entity.LivingEntity;
 import net.minecraft.util.Identifier;
 import net.minecraft.util.TypedActionResult;
 
+import java.util.ArrayList;
+
 public class BindingEvent {
 
-    public static final HandlerArray<NewBinding> NEW_BINDING = new HandlerArray<>(NewBinding.class);
-    public static final HandlerArray<GainSouls> GAIN_SOULS = new HandlerArray<>(GainSouls.class);
-    public static final HandlerArray<GetEntityName> GET_ENTITY_NAME = new HandlerArray<>(GetEntityName.class);
+    public static final ArrayList<NewBinding> NEW_BINDING = new ArrayList<>();
```

Comment:
This needs to be migrated to the new event api.

Reply:
I've made an attempt at changing this to the new event system; just committed it. I still don't fully understand how it functions, but the code works and is based off the Fabric API design. Still need to change the CageSpawnEvent.